epoch_test.go

```diff
@@ -170,6 +170,53 @@ func TestEpochConsecutiveProposalsDoNotGetVerified(t *testing.T) {
 	}
 }
 
+// TestEpochIncreasesRoundAfterFCert ensures that the epochs round is incremented
+// if we receive an fcert for the current round(even if it is not the next seq to commit)
+func TestEpochIncreasesRoundAfterFCert(t *testing.T) {
+	l := testutil.MakeLogger(t, 1)
+
+	bb := &testBlockBuilder{out: make(chan *testBlock, 1)}
+	storage := newInMemStorage()
+
+	wal := newTestWAL(t)
+
+	nodes := []NodeID{{1}, {2}, {3}, {4}, {5}, {6}}
+
+	conf := EpochConfig{
+		MaxProposalWait:     DefaultMaxProposalWaitTime,
+		Logger:              l,
+		ID:                  nodes[2],
+		Signer:              &testSigner{},
+		WAL:                 wal,
+		Verifier:            &testVerifier{},
+		Storage:             storage,
+		Comm:                noopComm(nodes),
+		BlockBuilder:        bb,
+		SignatureAggregator: &testSignatureAggregator{},
+	}
+
+	e, err := NewEpoch(conf)
+	require.NoError(t, err)
+
+	require.NoError(t, e.Start())
+
+	block, _ := advanceRoundFromNotarization(t, e, bb)
+	advanceRoundFromFinalization(t, e, bb)
+	require.Equal(t, uint64(2), e.Metadata().Round)
+	require.Equal(t, uint64(0), storage.Height())
```

Comment:
I don't understand why we would want to have a 0 height here. 

Shouldn't we expect the two blocks to commit?